### ISSUE TITLE
No Huber loss: MSE on a heavy-tailed 90-day target lets the tail drive gradients, rankings and accepts

### ISSUE THREAD

**Issue by nleck** [stSoftwareAU/NEAT-TS#3917]
## What is missing

`BUILT_IN_COST_NAMES` (`src/Costs.ts:23`) registers `MSE`, `RMSE`, `MAE`,
`MAPE`, `MSLE`, `HINGE`, `CROSS_ENTROPY`, with `CATEGORICAL_ERROR` added on the
Rust side. There is **no Huber loss** — grepping `src/costs/`,
`NEAT-AI-core/neat-core/src/loss.rs` and the scorer returns nothing.

That leaves a gap in the middle of the set we do have. `MSE` squares the
residual and is dominated by the tail; `MAE` treats a 10σ residual and a 0.1σ
residual as differing only linearly, and its gradient does not shrink near the
optimum. Huber is the standard answer: quadratic within `δ` of zero, linear
beyond it.

```
L(r) = ½r²                    for |r| ≤ δ
L(r) = δ(|r| − ½δ)            otherwise
```

Smooth and well-scaled near zero like MSE; bounded influence in the tail like
MAE.

## Why it matters for the production workload

The production creature predicts 90-day stock movement. That target is heavy
tailed — a small number of records carry residuals many times the typical
magnitude. Under MSE those records dominate three separate things:

1. **The gradient.** Backpropagation spends its capacity on the least
   predictable records in the corpus.
2. **The candidate ranking.** NEAT-AI-Discovery ranks by expected SSE reduction,
   so its detectors are drawn to the same extreme records. Whatever structure
   gets proposed is proposed to fit them.
3. **The accept/reject decision.** Every optimiser in the fleet accepts on
   full-corpus error. A candidate that fits three outlier records can beat one
   that improves ten thousand ordinary ones.

None of that is a bug. It is MSE doing exactly what MSE does, on a target shape
it suits badly.

## Proposed

Add `HUBER` as a first-class built-in cost, present on both sides of the wire.

**TypeScript** — `src/costs/HUBER.ts` implementing `CostInterface` alongside the
existing seven, registered in `src/Costs.ts`, with `δ` configurable and a
documented default.

**Rust** — the matching arm in `neat-core/src/loss.rs` and the scorer's
`cost.rs`, so `rust_scorer --cost HUBER` works. The scorer notes (`cli.rs`) that
`MSE`/`RMSE`/`MAE` are hosted on the GPU batched kernels and the rest fall back
to CPU; Huber may join the CPU set initially, which is fine — say so rather than
leaving it to be discovered.

**Choosing `δ`.** It is scale-dependent, so a hard-coded default is a trap.
Derive it from the residual distribution — a common choice is a fixed quantile
of `|r|` (e.g. the 90th percentile), computed once against the corpus and pinned
in configuration rather than recomputed per batch, so the loss stays comparable
across runs.

## The consequence that must be planned for

`score = 1 − error − complexityPenalty`. Changing the cost function changes
`error`, so **every score in the fleet is renumbered** and no historical score
remains comparable. That means:

- Huber runs as an **alternative** cost, opt-in, not a replacement for MSE.
- A creature scored under Huber must never be compared against one scored under
  MSE. The cost name belongs in the score provenance if it is not there already.
- The evaluation is a **parallel run**: same corpus, same creature, MSE vs Huber,
  and the question is not which produces a bigger number — that is meaningless —
  but which produces a creature that is better on the objective we actually care
  about.

Which raises the prerequisite: we do not currently have that objective written
down as a measurable quantity. Worth resolving alongside this.

## Related

- `docs/COST_FUNCTION_NOTES.md` in NEAT-AI-Discovery documents that its
  expected-improvement proxy is SSE and is exact only under MSE. A new cost needs
  an entry there, and the per-consumer audit that file already defines.
- stSoftwareAU/NEAT-AI-Discovery#2025 — the detector/citation pass, which touches
  the same cost-agnostic invariants.

## Acceptance

- [ ] `HUBER` in `BUILT_IN_COST_NAMES`, implemented in `src/costs/HUBER.ts`.
- [ ] Matching Rust implementation so `rust_scorer --cost HUBER` works, with GPU/CPU hosting stated explicitly.
- [ ] `δ` derived from the residual distribution and pinned in configuration, not hard-coded blind.
- [ ] Cost name recorded in score provenance so a Huber score can never be silently compared with an MSE one.
- [ ] `COST_FUNCTION_NOTES.md` entry in NEAT-AI-Discovery.
- [ ] A parallel MSE-vs-Huber run on the production creature, with the comparison made on something other than the raw score.

## Reference

Huber (1964), *Robust Estimation of a Location Parameter*.
